src/lib.rs

```diff
@@ -287,6 +292,17 @@ impl<T: Committable> RawCommitmentBuilder<T> {
         self
     }
 
+    pub fn optional<N, F>(self, o: &Option<N>, f: F) -> Self
+    where
+        N: Committable,
+        F: Fn(Self, &N) -> Self,
+    {
+        match o {
+            Some(obj) => f(self, obj),
```

Comment:
If this closure is provided by the user, it should be impossible to misuse. What if `f(self, obj)` just did `self.u64(0)`? We would have a collision between the Some and None cases (contrived example, I know). It would be better to add a specific domain separator (e.g. `self.u64(1)`)  before calling `f` so that no matter what `f` does, we can't have a collision.